### ISSUE TITLE
Launcher keeps deleting my saves and settings when updating.

### ISSUE THREAD

**Issue by Greevv** [remyroy/CDDA-Game-Launcher#314]
Title. Occasionally when I update the launcher will have wiped all my saves and settings, which as you can imagine, is a bit of a pain in the ass. It never did this before the most recent update.

**Comment by remyroy** [comment]
Have you checked in your recycle bin? The launcher moves stuff in there instead of deleting it since a few releases.

If you can find a reliable way to reproduce this issue, I could look into it.

**Comment by Greevv** [comment]
Yeah, I was able to recover everything, but it seems to be semi random as
to when it deletes my saves.

**Comment by Lorith** [comment]
From everything I have seen and heard, it occurs when a file it attempts to change has been opened by another program (which can be windows simply randomly looking at it in the background, hence why it appears random) -- I am not 100% certain on this, but it seems consistent with what I've heard over the years.

**Comment by remyroy** [comment]
We recent versions of the launcher, we are using the IFileOperation interface provided by Windows itself when deleting files. It should detect if a file is opened by an application and warn about it. We might need to use the same interface for moving files as well.

**Comment by remyroy** [comment]
With fdf9488b9f18fe7322a66b22dfefc48dee048648, we will also use the built in Windows File Operations for moving during the update's process backup part. It will warn if a file is opened by another application and it will ask to close that application to try again. It should help with this issue.

It will be part of the next release.
